src/hevm/hevm-cli/hevm-cli.hs

```diff
@@ -318,6 +362,107 @@ regexMatches regexSource =
   in
     Regex.matchTest regex . Seq.fromList . unpack
 
+equivalence :: Command Options.Unwrapped -> IO ()
+equivalence cmd =
+  do let bytecodeA = hexByteString "--code" . strip0x $ codeA cmd
+         bytecodeB = hexByteString "--code" . strip0x $ codeB cmd
+     maybeSignature <- case sig cmd of
+       Nothing -> return Nothing
+       Just sig' -> do method' <- functionAbi sig'
+                       return $ Just (view methodSignature method', snd <$> view methodInputs method')
+
+     void . runSMTWithTimeOut (solver cmd) (smttimeout cmd) . query $
+       equivalenceCheck bytecodeA bytecodeB (maxIterations cmd) maybeSignature >>= \case
+         Right vm -> do io $ putStrLn "Not equal!"
+                        io $ putStrLn "Counterexample:"
+                        showCounterexample vm maybeSignature
+                        io $ exitFailure
+         Left (postAs, postBs) -> io $ do
+           putStrLn $ "Explored: " <> show (length postAs)
+                       <> " execution paths of A and: "
+                       <> show (length postBs) <> " paths of B."
+           putStrLn $ "No discrepancies found."
+
+
+-- cvc4 sets timeout via a commandline option instead of smtlib `(set-option)`
+runSMTWithTimeOut :: Maybe Text -> Maybe Integer -> Symbolic a -> IO a
+runSMTWithTimeOut (Just "cvc4") Nothing sym  = runSMTWith cvc4 sym
+runSMTWithTimeOut (Just "cvc4") (Just n) sym = do
+  setEnv "SBV_CVC4_OPTIONS" ("--lang=smt --incremental --interactive --no-interactive-prompt --model-witness-value --tlimit-per=" <> show n)
+  a <- runSMTWith cvc4 sym
+  setEnv "SBV_CVC4_OPTIONS" ""
+  return a
+runSMTWithTimeOut smtsolver t sym =
+  let runwithz3 = runSMTWith z3 $ maybe (return ()) setTimeOut t >> sym
+  in case smtsolver of
+    Just "z3" -> runwithz3
+    Nothing   -> runwithz3
+    Just _ -> error "Unknown solver. Currently supported solvers; z3, cvc4"
+
+
+checkForVMErrors :: [EVM.VM] -> [String]
+checkForVMErrors [] = []
+checkForVMErrors (vm:vms) = case view EVM.result vm of
+  Just (EVM.VMFailure EVM.UnexpectedSymbolicArg) -> ("Unexpected symbolic argument at opcode: " <>
+                                             maybe "??" show (EVM.vmOp vm) <> ". Not supported (yet!)"):checkForVMErrors vms
+  _ -> checkForVMErrors vms
+
+-- Although it is tempting to fully abstract calldata and give any hints about the nature of the signature
+-- doing so results in significant time spent in consulting z3 about rather trivial matters. But with cvc4 it is quite pleasant!
+
+-- If function signatures are known, they should always be given for best results.
+assert :: Command Options.Unwrapped -> IO ()
+assert cmd = do
+
+  let root = fromMaybe "." (dappRoot cmd)
+      srcinfo = ((,) root) <$> (jsonFile cmd)
+      block'  = maybe EVM.Fetch.Latest EVM.Fetch.BlockNumber (block cmd)
+      rpcinfo = (,) block' <$> rpc cmd
+  maybesig <- case sig cmd of
+        Nothing -> return Nothing
+        Just sig' -> do method' <- functionAbi sig'
```

Comment:
`functionAbi` assumes that a solc > 0.6.7 is present in the users path (check `solidity'` in `Solidity.hs`), which is not currently the case on my system, meaning calls to `hevm symboilc` fail with:

```
Invalid option to --combined-json: storage-layout
```

maybe we can do something smart with `nix` here?

Reply:
like maybe a `makeWrapper` around `hevm` that places `solc` 0.6.7 in path for `hevm`?

Reply:
oh, wow, yeah, I thought it `solc` was fixed by nix here. I wonder if adding solc [here](https://github.com/dapphub/dapptools/blob/9d48a1a8b572654adad05a0218169f2b6beef9f2/haskell.nix#L57) would be sufficient

Reply:
oh yes, I think that should work :smiley_cat:

Reply:
hmm, looks like it doesn't...

Reply:
ok, with `--prefix` instead of `--suffix`, it seems to work